clang/include/clang/CConv/RewriteUtils.h

```diff
@@ -149,52 +139,54 @@ class ArrayBoundsRewriter {
   ProgramInfo &Info;
 };
 
+
 // Class for visiting declarations of variables and adding type annotations
 class TypeRewritingVisitor : public RecursiveASTVisitor<TypeRewritingVisitor> {
 public:
   explicit TypeRewritingVisitor(ASTContext *C, ProgramInfo &I,
-                                RSet &DR, std::set<std::string> &V,
-                                std::map<std::string, std::string> &newFuncSig,
+                                RSet &DR,
+                                std::map<std::string, std::string> &NewFuncSig,
                                 ArrayBoundsRewriter &ArrRewriter)
-          : Context(C), Info(I), rewriteThese(DR), VisitedSet(V),
-            ModifiedFuncSignatures(newFuncSig), ABRewriter(ArrRewriter) {}
+      : Context(C), Info(I), RewriteThese(DR), ABRewriter(ArrRewriter),
+        VisitedSet(), ModifiedFuncSignatures(NewFuncSig) {}
 
-  bool VisitCallExpr(CallExpr *);
   bool VisitFunctionDecl(FunctionDecl *);
   bool isFunctionVisited(std::string FuncName);
 private:
-  std::set<unsigned int> getParamsForExtern(std::string);
   // Get existing itype string from constraint variables.
   // if tries to get the string from declaration, however,
   // if there is no declaration of the function,
   // it will try to get it from the definition.
   std::string getExistingIType(ConstraintVariable *DeclC);
-  bool anyTop(CVarSet);
+
   ASTContext            *Context;
   ProgramInfo           &Info;
-  RSet                  &rewriteThese;
-  std::set<std::string> &VisitedSet;
-  std::map<std::string, std::string> &ModifiedFuncSignatures;
+  RSet                  &RewriteThese;
   ArrayBoundsRewriter   &ABRewriter;
+
+  // Set containing the names of all functions visited in the AST traversal.
+  // Used to ensure the new signature is only computed once for each function.
+  std::set<std::string> VisitedSet;
+
+  // This is a map from functions (the string representation of their names) to
+  // their function signature in the rewritten program.
+  std::map<std::string, std::string> &ModifiedFuncSignatures;
 };
 
 
 class RewriteConsumer : public ASTConsumer {
```

Comment:
Are there better names for these classes? What is `RewriteConsumer` and how does it relate to `TypeRewritingVisitor` in terms of their functional roles?

Reply:
It looks like the job of `TypeRewritting` visitor has changed significantly over time. One comment says it is "for visiting declarations of variables and adding type annotations." What it does right now is visit function declarations, figure out how to rewrite the declaration, and then store the rewritten declaration in a map without doing the actual rewriting.